crates/host/src/config/mod.rs

```diff
@@ -0,0 +1,167 @@
+use core::fmt;
+
+use std::collections::HashMap;
+use std::env;
+
+use serde::{de, Deserialize, Deserializer, Serialize};
+use url::Url;
+
+/// Cloud config
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Cloud {
+    #[serde(default)]
+    pub actors: HashMap<String, Actor>,
+    #[serde(default)]
+    pub links: Vec<Link>,
+}
+
+fn deserialize_actor_url<'de, D>(deserializer: D) -> Result<Url, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct Visitor;
+
+    impl<'de> de::Visitor<'de> for Visitor {
+        type Value = Url;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+            formatter.write_str("relative or absolute actor URL")
+        }
+
+        fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            match s.parse() {
+                Ok(url) => Ok(url),
+                Err(url::ParseError::RelativeUrlWithoutBase) => {
+                    let wd = env::current_dir().map_err(|e| {
+                        de::Error::custom(format!("failed to lookup current directory: {e}"))
+                    })?;
+                    Url::from_file_path(wd.join(s)).map_err(|()| {
+                        de::Error::custom(
+                            "failed to construct a `file` scheme URL for relative path".to_string(),
+                        )
+                    })
+                }
+                Err(e) => Err(de::Error::custom(format!(
+                    "failed to parse `{s}` as an actor URL: {e}"
+                ))),
+            }
+        }
+    }
+    deserializer.deserialize_str(Visitor)
+}
+
+/// Actor config
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(tag = "kind", deny_unknown_fields)]
+pub enum Actor {
+    #[serde(rename = "local")]
+    Local {
+        #[serde(deserialize_with = "deserialize_actor_url")]
+        url: Url,
+    },
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct TcpSocket {
+    pub addr: String, // TODO: Proper type
+}
+
+/// Link config
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(tag = "kind", deny_unknown_fields)]
+pub enum Link {
+    #[serde(rename = "interface")]
+    Interface {
+        name: String,
+        source: String,
+        target: String,
+    },
+    #[serde(rename = "tcp")]
+    Tcp {
```

Comment:
This might be related to Bailey's comment, but I'm not sure I understand what these enum values are. Is the `Interface` type for wasmCloud contract interfaces? And TCP is for local testing(?)

Reply:
This is used to define a TCP socket listener